Don't return module error when mysql_connect fails (#64560)

juergenhoetzel opened the pull request:
mysql_user expects an Exception when using check_implicit_admin.

##### SUMMARY

Fixes ##64560

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME

- mysql_user module
- mysql_utils

##### ADDITIONAL INFORMATION

Fixes regression bug #64560 introduced with #55434 (which raises `module.fail_json` in utility function `mysql_connect`)

juergenhoetzel committed:
Don't return module error when mysql_connect fails (#64560)

mysql_user expects an Exception when using check_implicit_admin.

Andersson007 reviewed (changes requested):
hello, @juergenhoetzel , thanks for fixing this
1. In the PR's title the ```postgresql_user``` module is mentioned but in the PR you changed ```mysql_info```
2. We need integration tests for this case which try to reproduce the situation when the wrong behavior occurs and check that all is ok after

juergenhoetzel committed:
Adds integration tests for mysql_user check_implicit_admin (#64560)

juergenhoetzel commented:
> 
> 1. In the PR's title the `postgresql_user` module is mentioned but in the PR you changed `mysql_info`

Cant confirm that: title is about mysql  :thinking: 
> 2. We need integration tests for this case which try to reproduce the situation when the wrong behavior occurs and check that all is ok after

I just added a test and rebased.

Andersson007 reviewed (approved):
LGTM, thanks!
shipit

nerijus commented:
Please apply to 2.9 branch too, as it worked with 2.8.

Andersson007 commented:
@nerijus , wanted to mention it, you were faster :) thanks!
@juergenhoetzel see [here](https://docs.ansible.com/ansible/latest/community/development_process.html#backporting-merged-prs) how to do it. If you can't/have no time, let me know, i'll backport it

Andersson007 commented:
if they won't merge it today, i'll ping somebody directly tomorrow morning

juergenhoetzel commented:
> if they don't merge it today, i'll ping somebody directly tomorrow morning

I'm not mad at you for doing this :smile:  I am already half asleep.

Andersson007 commented:
ok, sorry for the late conversation) they said we need a [changelog fragment](https://docs.ansible.com/ansible/latest/community/development_process.html#creating-a-changelog-fragment) in order to merge it

Andersson007 commented:
added the changelog fragment https://github.com/ansible/ansible/pull/65594

Andersson007 commented:
@juergenhoetzel morning! i made the backport https://github.com/ansible/ansible/pull/65596.
So you don't need to do anything else for solving this, thank you!
Thank you guys all!

juergenhoetzel commented:
> @juergenhoetzel morning! i made the backport #65596.
> So you don't need to do anything else for solving this, thank you!
> Thank you guys all!

@Andersson007 Thank you so much for taking care of it.

I wasn't aware of the extra stable branch process.

You must have had a short night :fearful:

Andersson007 commented:
@julienstroheker just i tried to make the night shorter, the ansible development is really suitable for that :)

misterunknown commented:
Thank you guys @juergenhoetzel @Andersson007 for the fast fix! Saved me a lot of work :heart_eyes:
